File: Java/LeetCode/BestTimetoBuyandSellStock.java
package Java.LeetCode;

class Solution {
    public int maxProfit(int[] prices) {
        int min = prices[0];
        int profit = 0;

        for (int p: prices) {
            min = Math.min(min, p);
            profit = Math.max(profit, p - min);
        }

        return profit;
    }
}

File: archive/python/BestTimeToBuyAndSellStock.py
def maxProfit(prices):
    buy, profit  = prices[0],0
    for sell in prices:
        if profit < sell - buy:
            profit = sell - buy

        if buy > sell:
            buy = sell

    return profit


prices = [10,1,5,6,7,1]
print(maxProfit(prices))